Source/Core/Core/HW/DVD/DVDMath.cpp

```diff
@@ -118,6 +124,34 @@ double CalculateSeekTime(u64 offset_from, u64 offset_to)
     return distance * LONG_SEEK_VELOCITY_INVERSE + LONG_SEEK_CONSTANT;
 }
 
+// Returns the time in seconds it takes for the disc to spin to the angle where the
+// read head is over the given offset, starting from the given time in seconds.
+double CalculateRotationalLatency(u64 offset, double time, bool wii_disc)
+{
+  // The data track on the disc is modelled as an Archimedean spiral.
+
+  // We assume that the inserted disc is spinning constantly, which
+  // is not always true (especially not when no disc is inserted),
+  // but doesn't lead to any problems in practice. It's as if every
+  // newly inserted disc is inserted at such an angle that it ends
+  // up rotating identically to a disc inserted at boot.
+
+  // To make the calculations simpler, the angles go from 0 to 1 rather
+  // than using radians. If you wish, you can replace every 1 with
+  // MathUtil::TAU in your head when reading the code. (This is why the
+  // code says * 1 and / 1 occasionally even though it has no effect.)
+
+  const double rotations_per_second = wii_disc ? WII_ROTATIONS_PER_SECOND : GC_ROTATIONS_PER_SECOND;
+  const double target_angle = fmod(CalculatePhysicalDiscPosition(offset) / DVD_TRACK_PITCH * 1, 1);
```

Comment:
Thoughts on `constexpr`-ing this `1` for the sanity of the reader, as...dunno, `max_angle`, `full_turn`, `full_rotation` or something along those lines?

Reply:
Done.